src/containers/setContacts/index.tsx

```diff
@@ -0,0 +1,202 @@
+    dispatch(resetSetContactState());
+  };
+
+  const onFinish = async (values: ContactFormFields) => {
+    if (!asyncRequestIsComplete(contacts)) {
+      throw new Error(
+        'Setting Form values requires knowing current contact information',
+      );
+    }
+
+    dispatch(
+      setContactInfo({
+        mainContact: {
+          email: contacts.result.mainContact.email,
+          firstName: values.firstName,
+          lastName: values.lastName,
+          phoneNumber: values.phoneNumber,
+          dateOfBirth: values.dateOfBirth.toDate(),
+          pronouns: values.pronouns || contacts.result.mainContact.pronouns,
+          allergies: values.allergies || contacts.result.mainContact.allergies,
+          diagnosis: values.diagnosis || contacts.result.mainContact.diagnosis,
+          medications:
+            values.medications || contacts.result.mainContact.medications,
+          notes: values.notes || contacts.result.mainContact.notes,
+          photoRelease: contacts.result.mainContact.photoRelease,
+          referrer: contacts.result.mainContact.referrer,
+          profilePicture: values.profilePicture
+            ? await encodeProfileFieldFile(values.profilePicture)
+            : contacts.result.mainContact.profilePicture,
+        },
+        additionalContacts: values.additionalContacts
+          ? await Promise.all(
+              values.additionalContacts.map(async (c, i) => ({
+                ...c,
+                dateOfBirth: c.dateOfBirth.toDate(),
+                profilePicture: c.profilePicture
+                  ? await encodeProfileFieldFile(c.profilePicture)
+                  : selectProfilePictureOrDefaultToUndefined(
+                      i,
+                      contacts.result.additionalContacts,
+                    ),
+              })),
+            )
+          : [],
+        children: values.children
+          ? await Promise.all(
+              values.children.map(async (c, i) => ({
+                ...c,
+                dateOfBirth: c.dateOfBirth.toDate(),
+                profilePicture: c.profilePicture
+                  ? await encodeProfileFieldFile(c.profilePicture)
+                  : selectProfilePictureOrDefaultToUndefined(
+                      i,
+                      contacts.result.children,
+                    ),
+              })),
+            )
+          : [],
+        location: {
+          address: values.address,
+          city: values.city,
+          state: values.state,
+          zipCode: values.zipCode,
+        },
+      }),
+    );
+  };
+
+  const contactsToFormFields = (info: ContactInfo): ContactFormFields => {
+    return {
+      ...info.location,
+      ...info.mainContact,
+      dateOfBirth: moment(new Date()),
+      profilePicture: undefined,
+      additionalContacts:
+        info.additionalContacts &&
+        info.additionalContacts.map((c) => ({
+          ...c,
+          dateOfBirth: moment(new Date()),
```

Comment:
imagine importing 290k of code for date utility functions 😔

Reply:
We gotta get rid of Momentjs man...

Reply:
agreed